sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/WritableContent.java

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.storage.blob.transfer;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.UriPermission;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+
+import androidx.annotation.MainThread;
+
+import java.io.Closeable;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Package private.
+ *
+ * A type that describes a content in the device to which data can be written.
+ */
+final class WritableContent {
+    private final Context context;
+    private final Uri contentUri;
+    private final boolean useContentResolver;
+    // Channel to write to the Content if ContentResolver is required to resolve the Content,
+    // i.e. when useContentResolver == true
+    private WriteToContentChannel contentChannel;
+
+    /**
+     * Create WritableContent describing a content in the device on which data can be written.
+     *
+     * @param context the context
+     * @param contentUri the content URI identifying the content
+     * @param useContentResolver indicate whether to use {@link android.content.ContentResolver} to resolve
+     *                           the content URI
+     */
+    WritableContent(Context context, Uri contentUri, boolean useContentResolver) {
+        this.context = context;
+        this.contentUri = contentUri;
+        this.useContentResolver = useContentResolver;
+    }
+
+    /**
+     * Get the {@link Uri} to the content.
+     *
+     * @return the content URI
+     */
+    Uri getUri() {
+        return this.contentUri;
+    }
+
+    /**
+     * Check whether to use {@link android.content.ContentResolver} to resolve the content URI.
+     *
+     * @return true if resolving content URI requires content resolver.
+     */
+    boolean isUseContentResolver() {
```

Comment:
See comment above about the naming of this method on `ReadableContent`.

Reply:
I guess you mean `isUsingContentResolver`, will change it, thanks!